lib/inference/hugging_face_text_generation.rb

```diff
@@ -22,11 +22,6 @@ def self.perform!(
         raise CompletionFailed if model.blank?
 
         url = URI(SiteSetting.ai_hugging_face_api_url)
-        if block_given?
```

Comment:
Moving to headers instead of path makes it compatible with both self-hosting and hosted API